File: codegen-gradle-plugin/src/main/java/io/vrap/rmf/codegen/plugin/RamlCodegenPlugin.java
package io.vrap.rmf.codegen.plugin;


import org.gradle.api.Plugin;
import org.gradle.api.Project;

import java.io.File;
import java.nio.file.Paths;

public class RamlCodegenPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {

        RamlCodeGeneratorTask task = project.getTasks().create("generateRamlStub", RamlCodeGeneratorTask.class);
    }


}

File: codegen-gradle-plugin/src/main/kotlin/io/vrap/rmf/codegen/plugin/RamlCodegenPlugin.kt
@file:JvmName("RamlCodegenPlugin")

package io.vrap.rmf.codegen.plugin

import org.gradle.api.NamedDomainObjectContainer
import org.gradle.api.Plugin
import org.gradle.api.Project

open class RamlCodegenPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        setupProject(project)
        createTasks(project)
    }

    fun setupProject(project: Project){
        val generators : NamedDomainObjectContainer<RamlGenerator>  =  project.container(RamlGenerator::class.java)
        generators.all { generator: RamlGenerator?->   generator?.targets = project.container(Target::class.java)}
        project.extensions.add("RamlGenerator", generators)
    }

    fun createTasks(project: Project){
        project.tasks.create("generateRamlStub",RamlCodeGeneratorTask::class.java)
    }
}
